File: 3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int i = 0, j = 0;
        int n = s.length();
        if(n == 1)return 1;
        map<char, int> m;
        int sol = 0;
        while(j < s.length()){
            if(m[s[j]]>0){
                while(s[i] != s[j]){
                    m[s[i]]--;
                    i++;
                }
                i++;
                j++;
            }else{
                m[s[j]]++;
                j++;
            }
            sol = max(sol, j-i);
        }
        return sol;
    }
};

File: Python/3.py
class Solution:
    def lengthOfLongestSubstring(self, nums: str) -> int:
        l = 0
        ss = set()
        sol = 0
        curr = 0
        for r in range(len(nums)):
            curr += 1
            while nums[r] in ss:
                curr-=1
                ss.remove(nums[l])
                l+=1
            ss.add(nums[r])
            sol = max(sol, r-l+1)
        return sol
